core/primitives/src/utils.rs

```diff
@@ -85,13 +82,42 @@ pub fn to_string_value(s: String) -> StringValue {
     res
 }
 
-pub fn proto_to_result<T>(proto: SingularPtrField<T>) -> Result<T, String> {
-    proto.into_option().ok_or_else(|| "Bad Proto".to_string())
+pub fn proto_to_result<T>(proto: SingularPtrField<T>) -> Result<T, Box<std::error::Error>> {
+    proto.into_option().ok_or_else(|| "Bad Proto".into())
 }
 
-pub fn proto_to_type<T, U>(proto: SingularPtrField<T>) -> Result<U, String>
+pub fn proto_to_type<T, U>(proto: SingularPtrField<T>) -> Result<U, Box<std::error::Error>>
 where
-    U: TryFrom<T, Error = String>,
+    U: TryFrom<T, Error = Box<std::error::Error>>,
 {
     proto_to_result(proto).and_then(TryInto::try_into)
 }
+
+/// A wrapper around Option<T> that provides native Display trait.
+/// Simplifies propagating automatic Display trait on parent structs.
+pub struct DisplayOption<T>(pub Option<T>);
+
+impl<T: fmt::Display> fmt::Display for DisplayOption<T> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self.0 {
+            Some(ref v) => write!(f, "Some({})", v),
+            None => write!(f, "None"),
+        }
+    }
+}
+
+impl<T> DisplayOption<T> {
+    pub fn into(self) -> Option<T> {
```

Comment:
This should be a part of:
```
impl From<DisplayOption<T>> for Option<T> 
```

Reply:
Outside of crate with Option.

Reply:
I do not understand what do you mean.

Reply:
It does not have to be inside Option crate. It is normal to implement traits for types we don't own, as long as we own DisplayOption